src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -50,28 +50,32 @@ object HomeworkSpecification extends Properties("Homework"):
   }
 
   property("negation") = forAll { (rational: Rational) =>
-    ???
+    new Rational(rational.numer * -1, rational.denom) == -rational
   }
 
   property("addition") = forAll { (left: Rational, right: Rational) =>
-    ???
+    left + right == new Rational(left.numer * right.denom + right.numer * left.denom, left.denom * right.denom)
   }
 
   property("subtraction") = forAll { (left: Rational, right: Rational) =>
-    ???
+    left - right == new Rational(left.numer * right.denom - right.numer * left.denom, left.denom * right.denom)
   }
 
   property("multiplication") = forAll { (left: Rational, right: Rational) =>
-    ???
+    left * right == new Rational(left.numer * right.numer, left.denom * right.denom)
   }
 
   property("division") = forAll { (left: Rational, numer: Int, denom: Int) =>
     val right = Rational(if numer == 0 then 1 else numer, abs(denom) + 1)
-    ???
+    if right.numer > 0 then left / right == new Rational(left.numer * right.denom, left.denom * right.numer)
+    else left / right == new Rational(left.numer * right.denom * -1, left.denom * right.numer * -1)
   }
 
-  property("division by zero") = forAll { (left: Rational, int: Int) =>
-    ???
-  }
+//I cannot check division by zero because after the first iteration
+// an exception is thrown about the impossibility of division by zero)
+
+/*  property("division by zero") = forAll { (left: Rational, int: Int) =>

```

Comment:
Please review the tests I provided for topics and find the solution how to test it.